### PR TITLE
Brand new stasis cryobags!

### PR DESCRIPTION
**Pull request by outbools**
Reverts #7793 
Fixes shitcode https://github.com/ParadiseSS13/Paradise/commit/38655f81dab140949d00413638e9c91af752e94d
Moves stasis `used` var check from `Life()` to bag code
Removes blasphemy which is relative pathing from bodybag file
Makes stasis bag technology make more sense:
- ID lock is werks
- you can freeze any living mob now
- this is true stasis now. **Every** living mob `life` handle proc in existense is stopped.
- It activates only on mobs in really bad shape - unconscious and dead - to make this thing more emergency saving tool and less slowly killing trap device.
- Also it werks only for 10 minutes so it's really not as good for traitor as you thought

For good reasons coder from ancient times decided it would be fine that you can lock stasis bag, however he doesn't even test it properly. Lock considered only ID and PDA for opening, ignoring the fact you can actually open it with your bare hand.

Another coder decided it would be great to get the var in stasis bag code which gets affected directly from `Life()` proc...

And third coder, from modern times, decided there is to much *effort* to make this thing work and this is unmaintainable - so he removed it and offers no replacement for it.

***BUT I THINK IT'S A GOOD ITEM FOR PARAMEDIC***
This is an important part of my default paramedic loadout. Even if I use it once in 3 rounds, it's serves me good to save people from death. I don't need spacacillin each round but it's a part of my default loadout too. I hope someone share my opinion here.

Now stasis bags looks more *maintainable* and ID lock is working as intended like it should 3 years ago.

**Github is only for code related discussions!**
Discuss balance/usage things here - https://nanotrasen.se/forum/topic/11049-pr-7793-cryostasis-bag-removal/

:cl: Injazz
add: Stasis bags returned, and they now can freeze every living mob in existence!
tweak: Stasis bags activate freeze only on unconscious and dead mobs.
tweak: Stasis bags now halt any living activity.
fix: You now can properly lock stasis bag with your ID
/ :cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/human/life.dm [hunk 1]
```diff
@@ -1059,11 +1082,11 @@
 		makeSkeleton()
 		return //No puking over skeletons, they don't smell at all!
 
-	if(!isturf(loc))
-		return
 
 	for(var/mob/living/carbon/human/H in range(decaylevel, src))
 		if(prob(2))
+			if(istype(loc,/obj/item/bodybag))
```
Comment (outbools) [inline code/modules/mob/living/carbon/human/life.dm]: it was BODYBAG feature to handle decay at slower rate why you ever removed it

File: code/modules/mob/living/carbon/human/human_defines.dm [hunk 2]
```diff
@@ -67,8 +67,6 @@ var/global/default_martial_art = new/datum/martial_art
 
 	var/check_mutations=0 // Check mutations on next life tick
 
-	var/heartbeat = 0
```
Comment (outbools) [inline code/modules/mob/living/carbon/human/human_defines.dm]: will kill now

File: code/modules/mob/living/carbon/human/life.dm [hunk 3]
```diff
@@ -1,10 +1,20 @@
+/mob/living/carbon/human
+
+	var/pressure_alert = 0
+	var/prev_gender = null // Debug for plural genders
+	var/temperature_alert = 0
+	var/in_stasis = 0
+	var/exposedtimenow = 0
+	var/firstexposed = 0
+	var/heartbeat = 0
```
Comment (outbools) [inline code/modules/mob/living/carbon/human/life.dm]: will kill everything except in_stasis

File: code/modules/mob/living/carbon/human/human.dm [hunk 4]
```diff
@@ -32,6 +32,8 @@
 
 	create_reagents(330)
 
+	prev_gender = gender // Debug for plural genders
```
Comment (outbools) [inline code/modules/mob/living/carbon/human/human.dm]: will kill

File: code/modules/mob/living/carbon/alien/larva/life.dm [hunk 5]
```diff
@@ -1,3 +1,6 @@
+/mob/living/carbon/alien/larva
+	var/temperature_alert = 0
```
Comment (outbools) [inline code/modules/mob/living/carbon/alien/larva/life.dm]: will... kill?

File: code/modules/mob/living/carbon/alien/humanoid/life.dm [hunk 6]
```diff
@@ -3,6 +3,7 @@
 	toxins_alert = 0
 	fire_alert = 0
 	pass_flags = PASSTABLE
+	var/temperature_alert = 0
```
Comment (outbools) [inline code/modules/mob/living/carbon/alien/humanoid/life.dm]: will kill

File: code/modules/mob/living/carbon/human/human_defines.dm (hunk 2 above)
Comment (outbools) [inline code/modules/mob/living/carbon/human/human_defines.dm]: oh please revive REVIVE

**Comment by Bigtimetripod** [comment]
What is this? Summer of feature removal reverts? But seems this is a good idea. Instead of stomping a feature to the curb, fix it and make it better. +1... if it does work.

**Comment by outbools** [comment]
@Bigtimetripod this works better then ever considering ID lock feature actually working now

But i still need to apply some edits for PR to look more _maintainable_

**Comment by Bigtimetripod** [comment]
Yeap. I know on Colonial Marines, stasisbag play a very important role. Since they removed cloners and cryo to make it more difficult for medics, then they have a machine that recycles used statisbags. Maybe a idea to look into in the future also. 

Id locking is a godsend TBH. Cannot tell ye how many times i had graytide fuck everything up by opening lockers or crap like that in hallways that im dragging.

**Comment by outbools** [comment]
I found out that cryobags doesnt work correctly for a loooooooong time.
It stops only human procs from happening in life proc, but guess WHAT - there is also mob/living handles happening in human life(), so they left untouched.
Occupant could escape bag easily, which doesn't feels right with "stasis" thing.

So here's few features and tweaks in addition to fixes, which makes cryobags finally a good use.
- You can stasis every living mob now.
- Stasis will activate only if user is unconscious. I don't want this thing become a slowly killing trap so conscious can left stasis bag as they used to in a former approach.

**Comment by outbools** [comment]
2 hours of testing mobs are fine after a stasis - except the intended clone and brain damage.

@ParadiseSS13/commit-access ready for review

**Comment by EvadableMoxie** [comment]
>I know on Colonial Marines, stasisbag play a very important role

They work on Colonial Marines mainly due to the length of time to reach proper medical facilities being so much longer, and stabilizing people being so much harder (And possible to fuck up and make worse).  I don't think they are nearly as useful on paradise and probably never will be without some pretty big changes.

One way to make them more useful would be to have them pause the defib timer.  As it currently works, I believe defibs just check the time of death versus the current time to determine if someone is receivable.

**Comment by outbools** [comment]
@EvadableMoxie defib time is handled by Life()
And new tweaks got almost everything from Life() halted.

**Comment by pinatacolada** [comment]
this could be even more useful as a paramedic item, if it fits inside a backpack, maybe even stack with other cryobags

**Comment by outbools** [comment]
it fits inside a backpack you just should unwrap it until you need to use it

**Comment by pinatacolada** [comment]
yeah i was thinking of roller beds im dum

File: code/modules/mob/living/life.dm [hunk 7]
```diff
@@ -2,6 +2,9 @@
 	set invisibility = 0
 	set background = BACKGROUND_ENABLED
 
+	if(in_stasis && !stat)
```
Comment (outbools) [inline code/modules/mob/living/life.dm]: *ahem*
that's opposite of what i wanted to do

File: code/game/objects/items/bodybag.dm [hunk 8]
```diff
@@ -7,25 +7,26 @@
...
-		..()
-		new /obj/item/bodybag(src)
-		new /obj/item/bodybag(src)
-		new /obj/item/bodybag(src)
-		new /obj/item/bodybag(src)
-		new /obj/item/bodybag(src)
-		new /obj/item/bodybag(src)
-		new /obj/item/bodybag(src)
+
+/obj/item/weapon/storage/box/New()
```
Comment (Citinited) [inline code/game/objects/items/bodybag.dm]: Wait, won't this spawn seven bodybags inside every new box?

**Comment by taukausanake** [comment]
@outbools I just want to say you are amazing. Not because of the PR, this is great in its own regard. Just reading everything you post is great

File: code/modules/mob/living/carbon/human/life.dm (hunk 1 above)
Comment (Citinited) [inline code/modules/mob/living/carbon/human/life.dm]: Yeah, just `return` here. No need for the `prob` check

File: code/game/objects/items/bodybag.dm [hunk 9]
```diff
@@ -89,4 +89,73 @@
...
+
+/obj/structure/closet/body_bag/cryobag/toggle(mob/user)
+	if(!opened && locked && !src.allowed(user))
+		to_chat(user, "<span class='warning'>Access denied. Medical access only.</span>")
+		return
+	..()
+
+/obj/structure/closet/body_bag/cryobag/MouseDrop(over_object, src_location, over_location)
+	if((over_object == usr && (in_range(src, usr) || usr.contents.Find(src))))
+		if(!ishuman(usr))	return
```
Comment (Citinited) [inline code/game/objects/items/bodybag.dm]: Put this `return` on a new line

File: code/game/objects/items/bodybag.dm [hunk 10]
```diff
@@ -89,4 +89,73 @@
...
+			continue
+		var/mob/living/M = AM
+		M.in_stasis = 0
+		used = 1	//We consider bag as used only if there is actual human in it
+	..()
+
+/obj/structure/closet/body_bag/cryobag/open()
+	. = ..()
+	if(used) //happens after dump_contents()
+		var/obj/item/O = new/obj/item(src.loc)
```
Comment (Citinited) [inline code/game/objects/items/bodybag.dm]: Are you putting a ghost item inside the cryobag just so it can't be folded up once used?

File: code/modules/mob/living/life.dm [hunk 11]
```diff
@@ -2,6 +2,9 @@
 	set invisibility = 0
 	set background = BACKGROUND_ENABLED
 
+	if(in_stasis && stat != CONSCIOUS)
```
Comment (Citinited) [inline code/modules/mob/living/life.dm]: `if(in_stasis && stat)`... though I'd probably just check for if they're inside a cryobag instead of giving all relevant mobs a whole var just for this

File: code/modules/mob/living/life.dm [hunk 12]
```diff
@@ -201,6 +204,16 @@
 			AdjustEarDamage(-0.05)
 			AdjustEarDeaf(-1)
 
+/mob/living/proc/handle_stasis()
```
Comment (Citinited) [inline code/modules/mob/living/life.dm]: This should be a human level proc

File: code/modules/mob/living/living_defines.dm [hunk 13]
```diff
@@ -51,6 +51,7 @@
 
 	var/list/surgeries = list()	//a list of surgery datums. generally empty, they're added when the player wants them.
 
+	var/in_stasis = 0
```
Comment (Citinited) [inline code/modules/mob/living/living_defines.dm]: See above - only humans should be able to be put in stasis

File: code/game/objects/items/bodybag.dm (hunk 8 above)
Comment (outbools) [inline code/game/objects/items/bodybag.dm]: good grief

File: code/game/objects/items/bodybag.dm (hunk 10 above)
Comment (outbools) [inline code/game/objects/items/bodybag.dm]: not inside - this spawns a ghost item INSTEAD of cryobag once used. Im not an inventor of this behaviour. This is what cryobags done before me

File: code/game/objects/items/bodybag.dm (hunk 8 above)
Comment (outbools) [inline code/game/objects/items/bodybag.dm]: yare yare

File: code/game/objects/items/bodybag.dm (hunk 9 above)
Comment (outbools) [inline code/game/objects/items/bodybag.dm]: yare yare daze

File: code/modules/mob/living/life.dm (hunk 11 above)
Comment (outbools) [inline code/modules/mob/living/life.dm]: it was done before and this was one of the reasons why it's get deleted

File: code/modules/mob/living/life.dm (hunk 12 above)
Comment (outbools) [inline code/modules/mob/living/life.dm]: no you can put every mob into bag now read PR desc

File: code/modules/mob/living/living_defines.dm (hunk 13 above)
Comment (outbools) [inline code/modules/mob/living/living_defines.dm]: It can be done with every living mob read the PR desc

File: code/modules/mob/living/life.dm (hunk 12 above)
Comment (taukausanake) [inline code/modules/mob/living/life.dm]: I don't think it matters for simple animals, Then again I might enjoy putting a mulebot or Beepsky inside a bag

Comment (Citinited) [inline code/modules/mob/living/life.dm]: Fair enough, I assumed it would only work for humans, my bad

File: code/modules/mob/living/living_defines.dm (hunk 13 above)
Comment (Citinited) [inline code/modules/mob/living/living_defines.dm]: Serves me right for not properly reading it xD

File: code/game/objects/items/bodybag.dm (hunk 10 above)
Comment (Citinited) [inline code/game/objects/items/bodybag.dm]: Ah, right.

File: code/game/objects/items/bodybag.dm [hunk 14]
```diff
@@ -7,25 +7,26 @@
...
-		new /obj/item/bodybag(src)
-		new /obj/item/bodybag(src)
-		new /obj/item/bodybag(src)
-		new /obj/item/bodybag(src)
-		new /obj/item/bodybag(src)
-		new /obj/item/bodybag(src)
+
+/obj/item/weapon/storage/box/bodybags/New()
+	..()
+	new /obj/item/bodybag(src)
```
Comment (FalseIncarnate) [inline code/game/objects/items/bodybag.dm]: You could probably make this a loop like
```
for(var/i = 0 to num_bags)
   new /obj/item/bodybag(src)
```
to make this more condensed and also easier to adjust in the future

File: code/game/objects/items/bodybag.dm [hunk 15]
```diff
@@ -40,47 +41,49 @@
...
+			src.name += t
+			src.overlays += image(src.icon, "bodybag_label")
+		else
+			src.name = "body bag"
+	//..() //Doesn't need to run the parent. Since when can fucking bodybags be welded shut? -Agouri
+		return
+	else if(istype(W, /obj/item/weapon/wirecutters))
+		to_chat(user, "You cut the tag off the bodybag")
+		src.name = "body bag"
+		src.overlays.Cut()
```
Comment (FalseIncarnate) [inline code/game/objects/items/bodybag.dm]: `src.` is not needed here, it is implied

File: code/game/objects/items/bodybag.dm [hunk 16]
```diff
@@ -40,47 +41,49 @@
...
-			to_chat(user, "You cut the tag off the bodybag")
-			src.name = "body bag"
-			src.overlays.Cut()
+		if(!in_range(src, user) && src.loc != user)
 			return
+		t = sanitize(copytext(t,1,MAX_MESSAGE_LEN))
+		if(t)
+			src.name = "body bag - "
+			src.name += t
+			src.overlays += image(src.icon, "bodybag_label")
```
Comment (FalseIncarnate) [inline code/game/objects/items/bodybag.dm]: `src.` is not needed here, it is implied

File: code/game/objects/items/bodybag.dm [hunk 17]
```diff
@@ -40,47 +41,49 @@
...
+/obj/structure/closet/body_bag/MouseDrop(over_object, src_location, over_location)
+	..()
+	if((over_object == usr && (in_range(src, usr) || usr.contents.Find(src))))
+		if(!ishuman(usr))
 			return
+		if(opened)
+			return 0
+		if(contents.len)
+			return 0
+		visible_message("[usr] folds up the [src.name]")
```
Comment (FalseIncarnate) [inline code/game/objects/items/bodybag.dm]: `src.` is not needed here, it is implied

File: code/game/objects/items/bodybag.dm [hunk 18]
```diff
@@ -89,4 +92,74 @@
...
+			continue
+		var/mob/living/M = AM
+		M.in_stasis = 0
+		used = 1	//We consider bag as used only if there is actual human in it
+	..()
+
+/obj/structure/closet/body_bag/cryobag/open()
+	. = ..()
+	if(used) //happens after dump_contents()
+		var/obj/item/O = new/obj/item(src.loc)
```
Comment (FalseIncarnate) [inline code/game/objects/items/bodybag.dm]: `src.` is not needed here, it is implied

File: code/game/objects/items/bodybag.dm [hunk 19]
```diff
@@ -89,4 +92,74 @@
...
+		O.name = "used stasis bag"
+		O.icon = src.icon
```
Comment (FalseIncarnate) [inline code/game/objects/items/bodybag.dm]: `src.` is not needed here, it is implied

File: code/game/objects/items/bodybag.dm [hunk 20]
```diff
@@ -89,4 +92,74 @@
...
+/obj/structure/closet/body_bag/cryobag/close()
+	. = ..()
+	for(var/atom/movable/AM in src) //Atom already forcemoved into src in parent close() so we just searching from contents of src
+		if(!ismob(AM))
+			continue
+		var/mob/living/M = AM
+		M.in_stasis = 1
+
+/obj/structure/closet/body_bag/cryobag/toggle(mob/user)
+	if(!opened && locked && !src.allowed(user))
```
Comment (FalseIncarnate) [inline code/game/objects/items/bodybag.dm]: `src.` is not needed here, it is implied

File: code/game/objects/items/bodybag.dm [hunk 21]
```diff
@@ -89,4 +92,74 @@
...
+		if(!ishuman(usr))
+			return
+		to_chat(usr, "<span class='warning'>You can't fold that up anymore..</span>")
+	..()
+
+/obj/structure/closet/body_bag/cryobag/attackby(W as obj, mob/user as mob, params)
+	if(istype(W, /obj/item/weapon/card/id) || istype(W, /obj/item/device/pda))
+		if(src.allowed(user))
+			src.locked = !src.locked
+			to_chat(user, "The controls are now [src.locked ? "locked. Medical access only." : "unlocked."]")
```
Comment (FalseIncarnate) [inline code/game/objects/items/bodybag.dm]: `src.` is not needed here, it is implied

File: code/game/objects/items/bodybag.dm [hunk 22]
```diff
@@ -89,4 +92,74 @@
...
+			to_chat(user, "<span class='notice'>It won't budge!</span>")
+
+/obj/item/bodybag/cryobag
+	name = "stasis bag"
+	desc = "A folded, non-reusable bag designed for the preservation of an occupant's body by stasis. Activates on occupant's unconscious state. Can be locked with medical ID."
+	icon = 'icons/obj/cryobag.dmi'
+	icon_state = "bodybag_folded"
+
+/obj/item/bodybag/cryobag/attack_self(mob/user)
+	var/obj/structure/closet/body_bag/cryobag/R = new /obj/structure/closet/body_bag/cryobag(user.loc)
```
Comment (FalseIncarnate) [inline code/game/objects/items/bodybag.dm]: this should probably be `get_turf(user)` instead of `user.loc`

File: code/game/objects/items/bodybag.dm [hunk 23]
```diff
@@ -7,25 +7,26 @@
 	icon_state = "bodybag_folded"
 	w_class = WEIGHT_CLASS_SMALL
 
-	attack_self(mob/user)
-		var/obj/structure/closet/body_bag/R = new /obj/structure/closet/body_bag(user.loc)
-		R.add_fingerprint(user)
-		qdel(src)
+/obj/item/bodybag/attack_self(mob/user)
+	var/obj/structure/closet/body_bag/R = new /obj/structure/closet/body_bag(user.loc)
```
Comment (FalseIncarnate) [inline code/game/objects/items/bodybag.dm]: this should probably be `get_turf(user)` instead of `user.loc`

File: code/game/objects/items/bodybag.dm (hunk 14 above)
Comment (Fox-McCloud) [inline code/game/objects/items/bodybag.dm]: `for(var/i in 1 to num_bags)` 

is the proper format.

**Comment by Ralta** [comment]
When I asked about the stasis bag removal, snowflakey code was the primary reason given. Glad to see it's coming back in a way that's actually improved over the original design! 👍

File: code/game/objects/items/bodybag.dm (hunk 22 above)
Comment (outbools) [inline code/game/objects/items/bodybag.dm]: how about no

Comment (outbools) [inline code/game/objects/items/bodybag.dm]: If I'm locked in a locker I shouldn't be able to open a bag outside the locker

File: code/game/objects/items/bodybag.dm (hunk 23 above)
Comment (outbools) [inline code/game/objects/items/bodybag.dm]: nu

**Comment by outbools** [comment]
Added processing! Now cryobag runs out of stasis juice after 10 minutes of work!

**Comment by outbools** [comment]
@Fox-McCloud reeeeady
@tigercat2000 steady
@Crazylemon64 go

**Comment by HugoLuman** [comment]
Neato!

File: code/game/objects/items/bodybag.dm [hunk 24]
```diff
@@ -1,33 +1,31 @@
...
 	icon = 'icons/obj/bodybag.dmi'
 	icon_state = "bodybag_folded"
 	w_class = WEIGHT_CLASS_SMALL
 
-	attack_self(mob/user)
-		var/obj/structure/closet/body_bag/R = new /obj/structure/closet/body_bag(user.loc)
-		R.add_fingerprint(user)
-		qdel(src)
+/obj/item/bodybag/attack_self(mob/user)
+	var/obj/structure/closet/body_bag/R = new /obj/structure/closet/body_bag(user.loc)
```
Comment (tigercat2000) [inline code/game/objects/items/bodybag.dm]: Please address this- As it is, this allows for opening bodybags inside places such as spacepods or mecha where they will become completely unrecoverable. Either check to make sure the user's location is a turf, or change this to a get_turf.

File: code/game/objects/items/bodybag.dm [hunk 25]
```diff
@@ -40,47 +38,49 @@
...
+		if(t)
+			name = "body bag - "
+			name += t
+			overlays += image(icon, "bodybag_label")
+		else
+			name = "body bag"
+	//..() //Doesn't need to run the parent. Since when can fucking bodybags be welded shut? -Agouri
+		return
+	else if(istype(W, /obj/item/weapon/wirecutters))
+		to_chat(user, "You cut the tag off the bodybag")
```
Comment (tigercat2000) [inline code/game/objects/items/bodybag.dm]: Add some kind of span classes here. And punctuation.

File: code/game/objects/items/bodybag.dm [hunk 26]
```diff
@@ -40,47 +38,49 @@
...
+		if(!in_range(src, user) && loc != user)
 			return
+		t = sanitize(copytext(t,1,MAX_MESSAGE_LEN))
+		if(t)
+			name = "body bag - "
+			name += t
+			overlays += image(icon, "bodybag_label")
+		else
+			name = "body bag"
+	//..() //Doesn't need to run the parent. Since when can fucking bodybags be welded shut? -Agouri
```
Comment (tigercat2000) [inline code/game/objects/items/bodybag.dm]: Either put this at the end or remove it entirely. Preferably the latter.

File: code/game/objects/items/bodybag.dm [hunk 27]
```diff
@@ -40,47 +38,49 @@
...
+/obj/structure/closet/body_bag/MouseDrop(over_object, src_location, over_location)
+	..()
+	if((over_object == usr && (in_range(src, usr) || usr.contents.Find(src))))
+		if(!ishuman(usr))
 			return
+		if(opened)
+			return 0
+		if(contents.len)
+			return 0
+		visible_message("[usr] folds up the [name]")
```
Comment (tigercat2000) [inline code/game/objects/items/bodybag.dm]: Remove `the [name]` and replace it with `[src]`.

File: code/game/objects/items/bodybag.dm (hunk 22 above)
Comment (tigercat2000) [inline code/game/objects/items/bodybag.dm]: Please address this- As it is, this allows for opening cryobags inside places such as spacepods or mecha where they will become completely unrecoverable. Either check to make sure the user's location is a turf, or change this to a get_turf.

File: code/game/objects/items/bodybag.dm [hunk 28]
```diff
@@ -89,4 +89,94 @@
...
+			continue
+		var/mob/living/M = AM
+		M.in_stasis = 0
+		used = 1	//We consider bag as used only if there is actual human in it
+	..()
+
+/obj/structure/closet/body_bag/cryobag/open()
+	. = ..()
+	if(used) //happens after dump_contents()
+		var/obj/item/O = new/obj/item(loc)
```
Comment (tigercat2000) [inline code/game/objects/items/bodybag.dm]: This is absolutely atrocious. Make a proper typepath for used stasis bags.

File: code/game/objects/items/bodybag.dm [hunk 29]
```diff
@@ -89,4 +89,94 @@
...
+		O.name = "used stasis bag"
+		O.icon = icon
+		O.icon_state = "bodybag_used"
+		O.desc = "Pretty useless now.."
```
Comment (tigercat2000) [inline code/game/objects/items/bodybag.dm]: Either 1 dot or 3 dots, never two.

File: code/game/objects/items/bodybag.dm [hunk 30]
```diff
@@ -89,4 +89,94 @@
...
+/obj/structure/closet/body_bag/cryobag/toggle(mob/user)
+	if(!opened && locked && !allowed(user))
+		to_chat(user, "<span class='warning'>Access denied. Medical access only.</span>")
+		return
+	..()
+
+/obj/structure/closet/body_bag/cryobag/process()
+	if(timer > world.time)
+		return
+	visible_message("<span class='notice'>[bicon(src)][name] run out of power. Stasis field disabled.</span>")
```
Comment (tigercat2000) [inline code/game/objects/items/bodybag.dm]: bicon should be avoided where unnecessary. This classifies as unnecessary. This is also gramatically incorrect, change `[name] run out of power` to `[src] runs out of power.`

File: code/game/objects/items/bodybag.dm [hunk 31]
```diff
@@ -89,4 +89,94 @@
...
+	for(var/atom/movable/AM in src) //Atom already forcemoved into src in parent close() so we just searching from contents of src
```
Comment (tigercat2000) [inline code/game/objects/items/bodybag.dm]: I love how this loop or really close to it is duplicated in three different places.

File: code/game/objects/items/bodybag.dm [hunk 32]
```diff
@@ -89,4 +89,94 @@
...
+		var/mob/living/M = AM
+		M.in_stasis = 0
+	processing_objects.Remove(src)
+	used = 1
+
+/obj/structure/closet/body_bag/cryobag/MouseDrop(over_object, src_location, over_location)
+	if((over_object == usr && (in_range(src, usr) || usr.contents.Find(src))))
+		if(!ishuman(usr))
+			return
+		to_chat(usr, "<span class='warning'>You can't fold that up anymore..</span>")
```
Comment (tigercat2000) [inline code/game/objects/items/bodybag.dm]: This doesn't have a return after it, so you actually still fold it up... did you actually test this?

File: code/modules/mob/living/carbon/human/life.dm [hunk 33]
```diff
@@ -1059,19 +1059,19 @@
 		makeSkeleton()
 		return //No puking over skeletons, they don't smell at all!
 
-	if(!isturf(loc))
-		return
 
 	for(var/mob/living/carbon/human/H in range(decaylevel, src))
+		if(istype(loc,/obj/item/bodybag))
```
Comment (tigercat2000) [inline code/modules/mob/living/carbon/human/life.dm]: There's a reason all of this was behind the prob() check- It keeps down on processing costs. Also, this istype needs a space in it.

File: code/modules/mob/living/life.dm [hunk 34]
```diff
@@ -201,6 +204,16 @@
...
+	// Handle side effects from stasis
+	if(in_stasis)
```
Comment (tigercat2000) [inline code/modules/mob/living/life.dm]: Why does this check `in_stasis` if it can only be called when `is_stasis` is true?

File: code/modules/surgery/organs/blood.dm [hunk 35]
```diff
@@ -18,6 +18,9 @@
 /mob/living/carbon/human/handle_blood()
 	var/list/blood_data = get_blood_data(get_blood_id())//PROCCEPTION
 
+	if(in_stasis)
```
Comment (tigercat2000) [inline code/modules/surgery/organs/blood.dm]: You cannot bleed out regardless of your current stat if you are in a cryobag.

File: code/game/objects/items/bodybag.dm (hunk 24 above)
Comment (outbools) [inline code/game/objects/items/bodybag.dm]: like they used to be

File: code/game/objects/items/bodybag.dm (hunk 26 above)
Comment (outbools) [inline code/game/objects/items/bodybag.dm]: do it in another PR

File: code/game/objects/items/bodybag.dm (hunk 25 above)
Comment (outbools) [inline code/game/objects/items/bodybag.dm]: in another PR, i'm not an author of this and yes this is existing bodybag code

File: code/game/objects/items/bodybag.dm (hunk 32 above)
Comment (outbools) [inline code/game/objects/items/bodybag.dm]: yeah and no you can't

File: code/modules/mob/living/carbon/human/life.dm (hunk 33 above)
Comment (outbools) [inline code/modules/mob/living/carbon/human/life.dm]: uhhhhhhh

File: code/game/objects/items/bodybag.dm [hunk 36]
```diff
@@ -40,47 +38,49 @@
...
+/obj/structure/closet/body_bag/attackby(W as obj, mob/user as mob, params)
+	if(istype(W, /obj/item/weapon/pen))
+		var/t = input(user, "What would you like the label to be?", text("[]", name), null)  as text
+		if(user.get_active_hand() != W)
 			return
-		else if(istype(W, /obj/item/weapon/wirecutters))
-			to_chat(user, "You cut the tag off the bodybag")
-			src.name = "body bag"
-			src.overlays.Cut()
+		if(!in_range(src, user) && loc != user)
```
Comment (Citinited) [inline code/game/objects/items/bodybag.dm]: Checking for adjacency is probably better here

File: code/game/objects/items/bodybag.dm [hunk 37]
```diff
@@ -40,47 +38,49 @@
...
 			return
+		t = sanitize(copytext(t,1,MAX_MESSAGE_LEN))
+		if(t)
+			name = "body bag - "
```
Comment (Citinited) [inline code/game/objects/items/bodybag.dm]: `name = "body bag - [t]"`

File: code/game/objects/items/bodybag.dm [hunk 38]
```diff
@@ -40,47 +38,49 @@
...
+			overlays += image(icon, "bodybag_label")
+		else
+			name = "body bag"
+	//..() //Doesn't need to run the parent. Since when can fucking bodybags be welded shut? -Agouri
+		return
+	else if(istype(W, /obj/item/weapon/wirecutters))
+		to_chat(user, "You cut the tag off the bodybag")
+		name = "body bag"
+		overlays.Cut()
+		return
```
Comment (Citinited) [inline code/game/objects/items/bodybag.dm]: You have a trailing return here

File: code/game/objects/items/bodybag.dm (hunk 24 above)
Comment (FalseIncarnate) [inline code/game/objects/items/bodybag.dm]: Broken old code is not an excuse for broken new code.

Comment (taukausanake) [inline code/game/objects/items/bodybag.dm]: It's also not an excuse for removal when it could be improved, but yeah better to improve it further

File: code/game/objects/items/bodybag.dm (hunk 26 above)
Comment (taukausanake) [inline code/game/objects/items/bodybag.dm]: You're working with the code in there. Might as well do it as it stands

**Comment by outbools** [comment]
nah feel free to finish this PR. Im done playing Paradise so i don't interested in stasis bags anymore

**Comment by matt81093** [comment]
Well that's sad. Still, good work @outbools 
I might pick this up if someone else doesn't. A niche 10-minute single-use item that has better stasis would probably be amazing for any paramedic players.
